app/models/article.rb

```diff
@@ -146,7 +146,33 @@ class Article < ApplicationRecord
       .tagged_with(tag_name)
   }
 
-  scope :cached_tagged_with, ->(tag) { where("cached_tag_list ~* ?", "^#{tag},| #{tag},|, #{tag}$|^#{tag}$") }
+  scope :cached_tagged_with, lambda { |tag|
+    case tag
+    when String
+      where("cached_tag_list ~ ?", "[[:<:]]#{tag}[[:>:]]")
```

Comment:
Could you somehow explain `"[[:<:]]#{tag}[[:>:]]"` within a comment? I think this is arcane enough for a comment :D. I found this in PostgreSQL documentation:

> There are two special cases of bracket expressions: the bracket expressions [[:<:]] and [[:>:]] are constraints, matching empty strings at the beginning and end of a word respectively